### PR TITLE
Feat/form focus styles

### PR DESCRIPTION
**Pull request by ferreroltd**
This pr adds focus styles for all input form and fixes some style issues

### CODE REVIEW

File: packages/libs/react-ui/src/styles/vars.css.ts [hunk 1]
```diff
@@ -53,7 +53,8 @@ export const vars = createGlobalTheme(':root', {
   },
   shadows: {
     // TODO: Update to match design system
-    $1: `0px 1px 2px 0 $colors$gray40`,
+    $1: `0px 1px 0 0 ${colorPalette.$gray30}`,
+    $2: `0px 1px 0 0 ${colorPalette.$gray60}`,
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/styles/vars.css.ts:57]: What is the reason for adding these to the vars.css file?

Comment (ferreroltd) [reply packages/libs/react-ui/src/styles/vars.css.ts:57]: the reason is that we probably sould predefine our shadows here and reuse them, so i added one for each theme, i can also not do it and specify it manually in the Form css

Comment (eileenmguo) [reply packages/libs/react-ui/src/styles/vars.css.ts:57]: If it's used somewhere else as well, I think it make sense to add it here, but then it would probably have a more descriptive token name. In this case I think it is specific to the input components so i would define it there

**Review by eileenmguo — COMMENTED** [review]
bottom border looks too bright in dark mode 
image

File: packages/libs/react-ui/src/styles/vars.css.ts (hunk 1 above)
Comment (ferreroltd) [reply packages/libs/react-ui/src/styles/vars.css.ts:57]: yeah i was thinking on giving it shadow input related name

File: packages/libs/react-ui/src/components/Form/Form.css.ts [hunk 2]
```diff
@@ -1,20 +1,73 @@
...
+
+    outlineOffset: '2px',
     selectors: {
       [`${darkThemeClass} &`]: {
-        outline: `2px solid ${vars.colors.$gray60}`,
+        outline: `2px solid ${fallbackVar(
+          statusOutlineColor,
+          vars.colors.$blue60,
+        )}`,
       },
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Form/Form.css.ts:42]: Why do you need this dark theme version?

File: packages/libs/react-ui/src/components/Form/Form.css.ts [hunk 3]
```diff
@@ -1,20 +1,73 @@
...
+      [`${darkThemeClass} &`]: {
+        outline: `2px solid ${fallbackVar(statusColor, vars.colors.$gray60)}`,
+      },
+    },
+  },
+]);
+
+export const statusClass = style([
+  sprinkles({
+    borderRadius: '$sm',
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Form/Form.css.ts:27]: I don't think you need this style

File: packages/libs/react-ui/src/components/Form/Form.css.ts [hunk 4]
```diff
@@ -1,20 +1,73 @@
...
+  }),
+  {
+    outline: `2px solid ${fallbackVar(
+      statusOutlineColor,
+      vars.colors.$blue60,
+    )}`,
+
+    outlineOffset: '2px',
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Form/Form.css.ts:35]: I think you can move this outlineOffset style to the baseContainerClass and then it should be applied regardless of the state

File: packages/libs/react-ui/src/components/Form/Input/Input.stories.tsx [hunk 5]
```diff
@@ -87,17 +86,6 @@ const meta: Meta<IInputProps> = {
...
-      description:
-        'Width of the leading text. Defaults to the size of the text itself.',
-      control: {
-        type: 'select',
-      },
-      options: [
-        '- Omit this property to auto-size the leading text',
-        ...Object.keys(vars.sizes).map((key) => key as keyof typeof vars.sizes),
-      ],
-    },
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Form/Input/Input.stories.tsx:100]: Why did you remove this? This should still be possible to apply via the Input

File: packages/libs/react-ui/src/components/Form/Input/Input.tsx [hunk 6]
```diff
@@ -32,24 +30,17 @@ export interface IInputProps
 
 export const Input: FC<IInputProps> = forwardRef<HTMLInputElement, IInputProps>(
   function Input(
-    {
-      outlined,
-      leadingText,
-      leadingTextWidth,
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Form/Input/Input.tsx:38]: LeadingTextWidth should still be able to be applied via the InputProps as well. Additionally, it should override the leadingTextWidth provided by the context if both are configured

File: packages/libs/react-ui/src/components/Form/Form.css.ts [hunk 7]
```diff
@@ -1,20 +1,73 @@
...
+    selectors: {
+      [`${darkThemeClass} &`]: {
+        outline: `2px solid ${fallbackVar(statusColor, vars.colors.$gray60)}`,
+      },
+    },
+  },
+]);
+
+export const statusClass = style([
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Form/Form.css.ts:25]: I don't think you're using this class at all. Can it be removed?

File: packages/libs/react-ui/src/components/Form/InputWrapper/InputWrapper.context.ts [hunk 8]
```diff
@@ -0,0 +1,16 @@
+import type { vars } from '@theme/vars.css';
+
+import { createContext } from 'react';
+import type { Status } from '../Form.css';
+
+export type OpenSections = string[];
+
+interface IInputWrapperContext {
+  status?: Status;
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Form/InputWrapper/InputWrapper.context.ts:9]: I saw earlier there was a `FormFieldStatus` type. Should that be used here as well or are these different?

File: packages/libs/react-ui/src/components/Form/Input/Input.stories.tsx (hunk 5 above)
Comment (ferreroltd) [reply packages/libs/react-ui/src/components/Form/Input/Input.stories.tsx:100]: I thought this always came throught the wrapper, not as a prop as well, i can bring it back

File: packages/libs/react-ui/src/components/Form/Select/Select.tsx [hunk 9]
```diff
@@ -34,13 +35,16 @@ export const Select: FC<ISelectProps> = forwardRef<
...
+
   const Icon = icon && SystemIcon[icon];
   const ChevronDown = SystemIcon.ChevronDown;
 
   return (
     <div
       className={classNames(containerClass, {
         [containerClassDisabled]: disabled,
+        [statusClass]: status,
         [baseOutlinedClass]: outlined,
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Form/Select/Select.tsx:48]: Previously in the Input component I saw you were able to apply the status colors using the baseOutlinedClass. Can you do that here too?

Comment (ferreroltd) [reply packages/libs/react-ui/src/components/Form/Select/Select.tsx:48]: yes

Comment (ferreroltd) [reply packages/libs/react-ui/src/components/Form/Select/Select.tsx:48]: yes

File: packages/libs/react-ui/src/components/Form/InputWrapper/InputWrapper.context.ts (hunk 8 above)
Comment (ferreroltd) [reply packages/libs/react-ui/src/components/Form/InputWrapper/InputWrapper.context.ts:9]: we do export it as FormFieldStatus, inside the context of form is called status, i can rename for consistency

File: packages/libs/react-ui/src/components/Form/TextField/TextField.tsx [hunk 10]
```diff
@@ -11,12 +11,14 @@ export interface ITextFieldProps
 export const TextField: FC<ITextFieldProps> = ({
   disabled = false,
   inputProps,
+  status,
   ...rest
 }) => {
   const { id } = inputProps;
+  console.log(status);
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Form/TextField/TextField.tsx:18]: cleanup

File: packages/libs/react-ui/src/components/Form/InputWrapper/InputWrapper.context.ts (hunk 8 above)
Comment (ferreroltd) [reply packages/libs/react-ui/src/components/Form/InputWrapper/InputWrapper.context.ts:9]: yes will do, i think it makes more sense

File: packages/libs/react-ui/src/styles/vars.css.ts (hunk 1 above)
Comment (eileenmguo) [reply packages/libs/react-ui/src/styles/vars.css.ts:57]: Shall remove these variables completely since they aren't in the sprinkles file anymore?

**Comment by eileenmguo** [comment]
When there is a status selected, it doesn't seem like there is a difference between the focused state and non focused state

File: packages/libs/react-ui/src/components/Form/Form.css.ts [hunk 11]
```diff
@@ -1,20 +1,48 @@
...
-import { style } from '@vanilla-extract/css';
+import { createVar, fallbackVar, style } from '@vanilla-extract/css';
+
+export type FormFieldStatus = 'disabled' | 'positive' | 'warning' | 'negative';
+export const statusColor = createVar();
+export const statusOutlineColor = createVar();
 
 export const baseOutlinedClass = style([
+  {
+    outline: `2px solid ${fallbackVar(statusColor, vars.colors.$gray30)}`,
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Form/Form.css.ts:12]: I think you mean to use the statusOutlineColor

File: packages/libs/react-ui/src/components/Form/Input/Input.tsx [hunk 12]
```diff
@@ -35,21 +35,26 @@ export const Input: FC<IInputProps> = forwardRef<HTMLInputElement, IInputProps>(
...
       icon,
       rightIcon,
+      leadingTextWidth: propLeadingTextWidth,
       disabled = false,
       ...rest
     },
     ref,
   ) {
+    const { status, leadingTextWidth: contextLeadingTextWidth } =
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Form/Input/Input.tsx:46]: nit: I would call this wrapperLeadingTextWidth

**Review by eileenmguo — COMMENTED** [review]
Looking good so far! I like the reduction of code 😍

**Review by eileenmguo — APPROVED** [review]
Only one comment left to remove the shadows from sprinkles

**Review by salamaashoush — APPROVED** [review]
there are some stuff that could be better but I guess it is more of a general issue in the ui library and should be fixed with tokens,  applying dark theme styles per component

**Review by KristinaSpasevska — APPROVED** [review]

**Review by sanderlooijenga — APPROVED** [review]
